build/test.js

```diff
@@ -36,6 +36,10 @@ function runFrontendUnitTests(singleRun, doneFn) {
   };
 
   let server = new karma.Server(localConfig, function(failCount) {
+    if (failCount > 0) {
+      // Kill backend server and cluster, if running.
+      gulp.start('kill-all');
```

Comment:
This does not fix the issue in all other cases. E.g., when lint or CSS styles rule fail, gulp will still hang up.

The only way to fix this is to remove 'wait-for-cluster' task. Otherwise this problem will come back every now and then.


How about creating new versions of integration-test tasks that also have local-up-cluster as a dependency. And make local-up-cluster finish only when cluster is healthly. I believe this would work.

@cheld What do you think?